src/plugins/URLIndicatorPlugin/URLIndicator.js

```diff
@@ -21,77 +21,95 @@
-                self.state = CONNECTED;
-            }
-            function updateIndicator() {
-                $http.get(self.URLpath).then(handleResponse, handleError);
-            }
-            updateIndicator();
-            $interval(updateIndicator, self.interval, 0, false);
+            this.indicator = simpleIndicator;
+            this.setDefaultsFromOptions(options);
+            this.setIndicatorToState(PENDING);
+
+            this.fetchUrl();
+            setInterval(this.fetchUrl, this.interval);
         }
 
-        URLIndicator.prototype.getCssClass = function () {
-            return this.cssClass;
-        };
-        URLIndicator.prototype.getGlyphClass = function () {
-            return this.state.glyphClass;
-        };
-        URLIndicator.prototype.getText = function () {
-            switch (this.state) {
+        URLIndicator.prototype.setIndicatorToState = function (state) {
+            switch (state) {
                 case CONNECTED: {
-                    return this.label + " is connected";
+                    this.indicator.text(this.label + " is connected");
+                    this.indicator.description(this.label + " is online, checking status every " + this.interval + " milliseconds.");
+                    break;
                 }
                 case PENDING: {
-                    return "Checking status of " + this.label + " please stand by...";
+                    this.indicator.text("Checking status of " + this.label + " please stand by...");
+                    this.indicator.description("Checking status of " + this.label + " please stand by...");
+                    break;
                 }
                 case DISCONNECTED: {
-                    return this.label + " is offline";
+                    this.indicator.text(this.label + " is offline");
+                    this.indicator.description(this.label + " is offline, checking status every " + this.interval + " milliseconds");
+                    break;
                 }
             }
+
+            this.indicator.statusClass(state.statusClass);
         };
-        URLIndicator.prototype.getDescription = function () {
-            switch (this.state) {
-                case CONNECTED: {
-                    return this.label + " is online, checking status every " +
-                    this.interval + " milliseconds.";
-                }
-                case PENDING: {
-                    return "Checking status of " + this.label + " please stand by...";
-                }
-                case DISCONNECTED: {
-                    return this.label + " is offline, checking status every " +
-                    this.interval + " milliseconds";
-                }
-            }
+
+        URLIndicator.prototype.fetchUrl = function () {
+            this.get(this.URLpath)
+                .then(this.handleSuccess)
+                .catch(this.handleError);
+        };
+
+        URLIndicator.prototype.get = function (url) {
```

Comment:
You should just spy on $.ajax instead of exposing a method purely to support testing; this also makes the test unnecessarily asynchronous.

Reply:
This was not done to support testing. I've made [my feelings](https://github.com/nasa/openmct/pull/1870#issuecomment-387175530) clear about writing code just to support tests. It was done to wrap the asynchronous behavior of this class in the promises model for consistency with the rest of our codebase and with modern HTTP APIs (eg. fetch). That said, it's extra code for marginal benefit here, so I'll just use the callbacks.